server/services/team_service.py

```diff
@@ -176,10 +176,47 @@ def delete_team_project(team_id, project_id):
         project.delete()
 
     @staticmethod
-    def get_all_teams() -> TeamsListDTO:
-        query = Team.query.distinct()
+    def get_all_teams(
+        team_name_filter: str = None,
+        team_role_filter: str = None,
+        member_filter: int = None,
+        member_request_filter: int = None,
+        manager_filter: int = None,
+    ) -> TeamsListDTO:
+
+        query = db.session.query(Team).join(TeamMembers).join(ProjectTeams)
```

Comment:
maybe the problem is on that line. I noticed that only the team that was assigned to a project was on the results